include/gridtools/fn/unstructured2.hpp

```diff
@@ -31,17 +31,30 @@ namespace gridtools::fn {
             Sizes m_sizes;
         };
 
+        template <class Domain, class Offsets>
+        struct domain_with_offsets {
+            Domain m_domain;
+            Offsets m_offsets;
+        };
+
         template <class Tag, class NeighborTable>
         typename hymap::keys<Tag>::template values<NeighborTable> connectivity(NeighborTable const &nt) {
             return {nt};
         }
 
         template <class... Connectivities>
-        auto unstructured_domain(
-            std::size_t horizontal_size, std::size_t vertical_size, Connectivities const &... conns) {
+        auto unstructured_domain(int horizontal_size, int vertical_size, Connectivities const &...conns) {
```

Comment:
Do you have a specific idea? I just implemented the most common overloads required by the examples. It’s still easily possible to use integral constants using the other overload if required. I simplified things slightly and added some static assertions checking the neighbor table types.

Reply:
What I meant: currently for integral_constants, the other overload is a better match because there is no conversion involved, but in the other constructor the first 2 args have different meaning. Maybe we should just remove this constructor for now.

Reply:
Thanks for the clarification; I removed it.